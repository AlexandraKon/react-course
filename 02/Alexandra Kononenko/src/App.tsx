
import Homes from './component/Homes/Homes';
import SearchHomes from './component/Search/SearchHomes';
import {data} from './data/data';

function App() {
  return (
    <div className='app'>
      <SearchHomes homes={data} />
      <Homes/>
    </div>
  );
}

export default App;
