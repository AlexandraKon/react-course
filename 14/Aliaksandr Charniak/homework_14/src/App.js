import React from 'react'

import Board from './components/Board'

class App extends React.Component {
  render() {
    return (
      <div>
        <h1>Game of life</h1>
        <Board />
      </div>
    )
  }
}

export default App
