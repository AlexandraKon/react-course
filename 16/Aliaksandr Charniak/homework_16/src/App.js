import React from 'react'
import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom'

import Civilizations from './components/Civilizations'
import Units from './components/Units'
import Structures from './components/Structures'
import Technologies from './components/Technologies'

import './styles.css'

class App extends React.Component {
  render() {
    return (
      <Router>
        <nav className="navigation">
          <Link to="/civilizations">Цивилизации</Link>
          <Link to="/units">Юниты</Link>
          <Link to="/structures">Структуры</Link>
          <Link to="/technologies">Технологии</Link>
        </nav>
        <Switch>
          <Route path="/civilizations" exact component={Civilizations} />
          <Route path="/civilizations/:id/unique_unit/:unit_name" exact component={Units} />
          <Route path="/civilizations/:id/unique_tech/:tech_name" exact component={Technologies} />
          <Route path="/units" exact component={Units} />
          <Route path="/units/:id/created_in/:created_in" exact component={Structures} />
          <Route path="/structures" exact component={Structures} />
          <Route path="/technologies" exact component={Technologies} />
          <Route path="/technologies/:id/develops_in/:created_in" exact component={Structures} />
          <Route path="/technologies/:id/applies_to/unit/:unit_name" exact component={Units} />
          <Route
            path="/technologies/:id/applies_to/civilization/:unit_name"
            exact
            component={Civilizations}
          />
        </Switch>
      </Router>
    )
  }
}

export default App
