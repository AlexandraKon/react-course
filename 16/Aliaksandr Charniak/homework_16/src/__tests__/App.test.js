import { configure, shallow } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'

import App from '../App'

configure({ adapter: new Adapter() })

describe('App unit test', () => {
  const wrapper = shallow(<App />)

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render 1 <App /> component', () => {
    expect(wrapper).toHaveLength(1)
  })

  it('should correctly count Router', () => {
    expect(wrapper.find('BrowserRouter')).toHaveLength(1)
  })

  it('should correctly count Switch', () => {
    expect(wrapper.find('Switch')).toHaveLength(1)
  })

  it('should correctly count Link', () => {
    expect(wrapper.find('Link')).toHaveLength(4)
  })

  it('should correctly count Route', () => {
    expect(wrapper.find('Route')).toHaveLength(10)
  })
})
