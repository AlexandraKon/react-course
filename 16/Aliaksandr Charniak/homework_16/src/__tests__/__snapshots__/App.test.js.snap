// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App unit test should match snapshot 1`] = `
<BrowserRouter>
  <nav
    className="navigation"
  >
    <Link
      to="/civilizations"
    >
      Цивилизации
    </Link>
    <Link
      to="/units"
    >
      Юниты
    </Link>
    <Link
      to="/structures"
    >
      Структуры
    </Link>
    <Link
      to="/technologies"
    >
      Технологии
    </Link>
  </nav>
  <Switch>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      exact={true}
      path="/civilizations"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      exact={true}
      path="/civilizations/:id/unique_unit/:unit_name"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      exact={true}
      path="/civilizations/:id/unique_tech/:tech_name"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      exact={true}
      path="/units"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      exact={true}
      path="/units/:id/created_in/:created_in"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      exact={true}
      path="/structures"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      exact={true}
      path="/technologies"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      exact={true}
      path="/technologies/:id/develops_in/:created_in"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      exact={true}
      path="/technologies/:id/applies_to/unit/:unit_name"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
        }
      }
      exact={true}
      path="/technologies/:id/applies_to/civilization/:unit_name"
    />
  </Switch>
</BrowserRouter>
`;
